opta/layer.py

```diff
@@ -189,14 +191,20 @@ def structured_config(self) -> StructuredConfig:
         }
 
     @classmethod
-    def load_from_dict(cls, conf: Dict[Any, Any], env: Optional[str]) -> Layer:
+    def load_from_dict(
+        cls, conf: Dict[Any, Any], env: Optional[str], is_parent: bool = False
+    ) -> Layer:
         modules_data = conf.get("modules", [])
         environments = conf.pop("environments", None)
         original_spec = conf.pop("original_spec", "")
         path = conf["path"]
         name = conf.pop("name", None)
         if name is None:
             raise UserErrors("Config must have name")
+        if is_parent and environments is not None:
+            raise UserErrors(
+                f"Layer {name} can not be a parent as it has a parent of its own"
```

Comment:
Let's not use "Layer" and "parent" for user facing stuff. But environment and app/service.